components/brave_rewards/resources/request_ads_enabled/lib/host.ts

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Host, HostState } from './interfaces'
+import { createStateManager } from '../../shared/lib/state_manager'
+
+const rewardsTourURL = 'brave_rewards_panel.html#tour'
+
+export function createHost (): Host {
+  const port = chrome.runtime.connect({ name: 'request-enable-rewards-panel' })
+
+  const stateManager = createStateManager<HostState>({
+    loading: true,
+    rewardsEnabled: false
+  })
+
+  chrome.braveRewards.shouldShowOnboarding((showOnboarding) => {
+    stateManager.update({
+      loading: false,
+      rewardsEnabled: !showOnboarding
+    })
+  })
+
+  return {
+    get state () { return stateManager.getState() },
+
+    addListener: stateManager.addListener,
+
+    getString (key) {
+      return chrome.i18n.getMessage(key,
+        ['$1', '$2', '$3', '$4', '$5', '$6', '$7', '$8', '$9'])
+    },
+
+    openRewardsTour () {
+      window.location.href = rewardsTourURL
+    },
+
+    enableRewards () {
+      chrome.braveRewards.enableRewards()
```

Comment:
we should not be called `enableRewards` here. You're re-introducing the race condition we wanted to eliminate